pkg/synthetictests/disruption_test.go

```diff
@@ -0,0 +1,120 @@
+package synthetictests
+
+import (
+	"testing"
+	"time"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_dnsOverlapDisruption(t *testing.T) {
+	now := time.Now()
+
+	type args struct {
+		events monitorapi.Intervals
+	}
+	tests := []struct {
+		name string
+		args args
+		want []*junitapi.JUnitTestCase
+	}{
+		{
+			name: "Detect disruption just after DNS lookup errors",
+			args: args{
+				events: []monitorapi.EventInterval{
+					{
+						Condition: monitorapi.Condition{
+							Locator: "disruption/openshift-api connection/new",
+							Message: "reason/DisruptionSamplerOutageBegan DNS lookup timeouts began for disruption/openshift-api connection/new requests over new connections: Get \"https://api.ci-op-v352gg5m-99751.aws-2.ci.openshift.org:6443/apis/image.openshift.io/v1/namespaces/default/imagestreams\": dial tcp: lookup api.ci-op-v352gg5m-99751.aws-2.ci.openshift.org: i/o timeout (likely a problem in cluster running tests, not the cluster under test)",
+						},
+						From: now.Add(-31 * time.Second),
+						To:   now.Add(-30 * time.Second),
+					},
+					{
+						Condition: monitorapi.Condition{
+							Level:   0,
+							Locator: "disruption/cache-kube-api connection/new",
+							Message: "reason/DisruptionSamplerOutageBegan DNS lookup timeouts began for disruption/cache-kube-api connection/new GET requests over new connections: Get \"https://api.ci-op-v352gg5m-99751.aws-2.ci.openshift.org:6443/api/v1/namespaces/default?resourceVersion=0\": dial tcp: lookup api.ci-op-v352gg5m-99751.aws-2.ci.openshift.org: i/o timeout (likely a problem in cluster running tests, not the cluster under test)",
+						},
+						From: now.Add(-29 * time.Second),
+						To:   now.Add(-28 * time.Second),
+					},
+				},
+			},
+			want: []*junitapi.JUnitTestCase{
+				&junitapi.JUnitTestCase{
+					FailureOutput: &junitapi.FailureOutput{
+						Message: "reason/DisruptionSamplerOutageBegan DNS lookup timeouts began for disruption/openshift-api connection/new GET requests over new connections: Get \"https://api.ci-op-v352gg5m-99751.aws-2.ci.openshift.org:6443/apis/image.openshift.io/v1/namespaces/default/imagestreams\": dial tcp: lookup api.ci-op-v352gg5m-99751.aws-2.ci.openshift.org on 172.30.0.10:53: read udp 10.131.172.119:52726-\u003e172.30.0.10:53: i/o timeout (likely a problem in cluster running tests, not the cluster under test)",
```

Comment:
Based on your code, shouldn't this FailureOutput include: "These failures imply Disruption overlapped DNS failure" ? If so, how is this passing? Actually locally it looks like all three tests are failing. :)

Reply:
I suspect you did not make it to tests yet, just got the code updated, in which case this makes more sense. I'll leave notes on the tests as review regardless.